### ISSUE TITLE
gitlab: allow configuring workflow run trigger criteria

### ISSUE THREAD

**Issue by mvidalgarcia** [reanahub/reana-server#432]
**Current behavior**

Currently, once a user [enables the GitLab integration](https://docs.reana.io/advanced-usage/code-repositories/gitlab/) of one of their projects, a webhook is created. This webhook is configured in such a way that every time they [push a new commit](https://github.com/reanahub/reana-server/blob/ce486033fc4d745d6e3b18c5162351d7465ecd8a/reana_server/rest/gitlab.py#L314) in the [`master` branch](https://github.com/reanahub/reana-server/blob/ce486033fc4d745d6e3b18c5162351d7465ecd8a/reana_server/rest/gitlab.py#L315) or a [merge request is created](https://github.com/reanahub/reana-server/blob/ce486033fc4d745d6e3b18c5162351d7465ecd8a/reana_server/rest/gitlab.py#L316), a new REANA workflow run is triggered as long as [there's a `reana.yaml` spec](https://github.com/reanahub/reana-server/blob/ce486033fc4d745d6e3b18c5162351d7465ecd8a/reana_server/utils.py#L287) file in the repo root directory.

When the hook callback is called (POST `/api/workflows`), we [handle the request](https://github.com/reanahub/reana-server/blob/ce486033fc4d745d6e3b18c5162351d7465ecd8a/reana_server/utils.py#L288-L293) taking into account if it was a push or a merge request by downloading the appropriate branch in order to create and submit the workflow.

GitLab API Project Hooks reference: https://docs.gitlab.com/ee/api/projects.html#hooks

**Expected behavior**

Ideally, the user should be able to configure under which conditions the REANA workflow CI is triggered. We could improve the GitLab webhook creation to pass options like `tag_push_events` (Trigger hook on tag push events.) so the CI is only run on tag creation.

To summarize, there're 3 criteria that the user should be able to toggle:
- Push events (to `master`)
- Merge requests events
- Tag push events

The UI should be changed accordingly in order to allow this fine-grained configuration. A [Toggle](https://react.semantic-ui.com/addons/radio/#types-toggle) element, as we have now, is too limited. Perhaps we could keep the toggle to enable the integration and a cog icon that opens a modal with checkboxes to check/uncheck the other options. A brief UI mockup session is needed.

Ideas on how to represent this in the UI are welcome 🎨

**Comment by mdonadoni** [comment]
Feature requested here: https://forum.reana.io/t/reana-cern-gitlab-questions/80
